go-controller/pkg/factory/factory.go

```diff
@@ -534,3 +608,43 @@ func (wf *WatchFactory) AddNodeHandler(handlerFuncs cache.ResourceEventHandler,
 func (wf *WatchFactory) RemoveNodeHandler(handler *Handler) error {
 	return wf.removeHandler(nodeType, handler)
 }
+
+// GetPod returns the pod spec given the namespace and pod name
+func (oc *objectCache) GetPod(namespace, name string) (*kapi.Pod, error) {
```

Comment:
@abhat if we have the lister in the infomrer itself, then this function (and the others) should just become

```
func (wf *WatchFactory) GetPod(namespace, name string) (*kapi.Pod, error) {
	return wf.informers[podType].lister.Pods(namespace).Get(name)
}
```

Reply:
See https://github.com/ovn-org/ovn-kubernetes/pull/1034#discussion_r373653141

Reply:
@dcbw, lister is itself not an interface. Individual informers would need to have a specific lister type to realize what you are suggesting. Thoughts?

Reply:
@abhat yep, you're right. But we know exactly what `lister` should be for any given type. So we can make the `lister` member be type `interface{}` and then this becomes:

```
func (wf *WatchFactory) GetPod(namespace, name string) (*kapi.Pod, error) {
	podLister := wf.informers[podType].lister.(*kapi.PodLister)
	return podLister.Pods(namespace).Get(name)
}
```
Yeah, it's casting, but we control all the types, and IMHO a bunch less code and keeps things that are logically together together.